src/js/vdlx-datagrid/datagrid.js

```diff
@@ -273,7 +281,18 @@ class Datagrid {
             height: '100%',
             resizableColumns: false,
             dataFiltered: saveState,
-            dataSorting: saveState,
+            dataSorting: () => {
+                this.tableLock && this.tableLock.lock();
+                sortPromise = new Promise(resolve => {
+                    sortPromiseResolve = resolve;
+                });
+                perf('datagrid sorting', constant(sortPromise));
+                saveState();
```

Comment:
why do we need to save state before sorting?

Reply:
Because that was the previous behaviour. I've just added in table locking and a performance monitor to the same dataSorting hook.